### PR TITLE
Tiles/fix

### PR DESCRIPTION
**Pull request by LucaMarconato**
https://github.com/scverse/spatialdata/pull/206 should be merged first so the diff in this PR becomes minimarl.

### CODE REVIEW

**Comment by LucaMarconato** [comment]
Ready for review.

**Comment by LucaMarconato** [comment]
@giovp @kevinyamauchi  this PR fixes some bugs in the tiling code, please review and merge when you have time.

File: src/spatialdata/dataloader/datasets.py [hunk 1]
```diff
@@ -114,9 +117,18 @@ def __getitem__(self, idx: int) -> tuple[SpatialImage, str, int]:
...
-            centroid = np.atleast_2d(region.geometry.coords._coords[0])
+            shape = regions.geometry.iloc[0]
+            if isinstance(shape, Polygon):
+                centroid = np.atleast_2d(region.geometry.centroid.coords._coords[0])
+            elif isinstance(shape, MultiPolygon):
+                raise NotImplementedError("MultiPolygon not supported yet")
+            elif isinstance(shape, Point):
+                # the function coords.xy is just accessing _coords, and wrapping it with extra information,
+                # so we access it directly
+                centroid = np.atleast_2d(region.geometry.coords._coords[0])
```
Comment (kevinyamauchi) [inline src/spatialdata/dataloader/datasets.py:128]: Do we have a test to cover this to catch if they change the private API?

**Review by kevinyamauchi — APPROVED** [review]
Hey @LucaMarconato ! This looks good to me. I made a suggestion for a test to be added. Once you've done that, please feel free to merge.

File: src/spatialdata/dataloader/datasets.py (hunk 1 above)
Comment (LucaMarconato) [reply src/spatialdata/dataloader/datasets.py:128]: Good point! Seems to be already broken. I am replacing this with public APIs and adding some tests (so far the code is not tested).

Comment (LucaMarconato) [reply src/spatialdata/dataloader/datasets.py]: Turned out the private APIs were still working (and did something wrong with the tests). Anyway, I removed them and now we are using public APIs.

**Pull request merged**